File: Online judge/Boj/C++/10820.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);

	while (1) {
		string s;
		getline(cin, s);
		if (s.length() == 0) break;
		int small = 0, capital = 0, number = 0, space = 0;
		for (int i = 0; i < s.length(); i++) {
			if (65 <= s[i] && s[i] <= 90) {
				capital++;
			}
			else if (97 <= s[i] && s[i] <= 122) {
				small++;
			}
			else if (48 <= s[i] && s[i] <= 57) {
				number++;
			}
			else if (s[i] == 32) {
				space++;
			}
		}
		cout << small << ' ' << capital << ' ' << number << ' ' << space << '\n';
	}
}

File: Online judge/Boj/Python/10820.py
while True:
  try:
    lis = input()
    if len(lis) == 0: break
    capital, small, number, space = 0,0,0,0
    for x in lis:
      if 65 <= ord(x) and ord(x) <= 90:
        capital += 1
      elif 97 <= ord(x) and ord(x) <= 122:
        small += 1
      elif 48 <= ord(x) and ord(x) <= 57:
        number += 1
      elif ord(x) == 32:
        space += 1
    print(small,capital,number,space)
  except:
    break